prusti-server/src/backend.rs

```diff
@@ -0,0 +1,27 @@
+use prusti_common::{
+    vir::{LoweringContext, ToViper},
+    Stopwatch,
+};
+use viper::{VerificationContext, VerificationResult};
+
+pub enum Backend<'a> {
+    Viper(viper::Verifier<'a>, &'a VerificationContext<'a>),
+}
+
+impl<'a> Backend<'a> {
+    pub fn verify(&mut self, program: &prusti_common::vir::program::Program) -> VerificationResult {
+        match self {
+            Backend::Viper(viper, context) => {
+                let mut stopwatch = Stopwatch::start("prusti-server", "verifier startup");
+
+                let ast_factory = context.new_ast_factory();
+                let viper_program = program.to_viper(LoweringContext::default(), &ast_factory);
+
+                context.new_ast_utils().with_local_frame(16, || {
```

Comment:
Move `.with_local_frame(16, || {` earlier, so that also `let ast_factory = ` is inside the local frame. We use `with_local_frame` as a safeguard, to be sure that we let the JVM deallocate objects that we constructed while constructing the viper program or while verifying.